src/main/java/org/reaktivity/nukleus/http/internal/routable/stream/TargetInputEstablishedStreamFactory.java

```diff
@@ -465,15 +458,29 @@ private void decodeCompleteHttpBegin(
                 final Map<String, String> headers = decodeHttpHeaders(start, lines);
                 // TODO: replace with lightweight approach (end)
 
+                resolveTarget();
+
                 target.doHttpBegin(targetId, 0L, sourceCorrelationId,
                         hs -> headers.forEach((k, v) -> hs.item(i -> i.name(k).value(v))));
                 target.setThrottle(targetId, this::handleThrottle);
 
-                boolean hasUpgrade = headers.containsKey("upgrade");
+                boolean upgraded = "101".equals(headers.get(":status"));
+                String connectionOptions = headers.get("connection");
+                if (connectionOptions != null)
+                {
+                    Arrays.asList(connectionOptions.toLowerCase().split(",")).stream().forEach((element) ->
+                    {
+                        if (element.equals("close"))
+                        {
+                            clientConnectReplyState.connection.persistent = false;
+                        }
+                    });
+                }
 
                 // TODO: wait for 101 first
```

Comment:
TODO should be removed as we have already received the `101 Switching Protocols` at this point.